consensus/src/consensus/mod.rs

```diff
@@ -610,6 +615,97 @@ impl ConsensusApi for Consensus {
         sample_headers
     }
 
+    fn get_utxo_return_script_public_key(&self, txid: Hash, target_daa_score: u64) -> Option<ScriptPublicKey> {
+        // We need consistency between the past pruning points, selected chain and header store reads
+        let _guard = self.pruning_lock.blocking_read();
+
+        let sc_read = self.selected_chain_store.read();
+
+        let pp_hash = self.pruning_point_store.read().get().unwrap().pruning_point;
```

Comment:
if we get `pruning_point` / `history_root` daa score and it is more then `target_daa_score`, we could return early with `None` here.